sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AQEUtils.scala

```diff
@@ -28,16 +28,28 @@ object AQEUtils {
   def getRequiredDistribution(p: SparkPlan): Option[Distribution] = p match {
     // User-specified repartition is only effective when it's the root node, or under
     // Project/Filter/LocalSort/CollectMetrics.
-    // Note: we only care about `HashPartitioning` as `EnsureRequirements` can only optimize out
-    // user-specified repartition with `HashPartitioning`.
```

Comment:
```
as `EnsureRequirements` can only optimize out user-specified repartition with `HashPartitioning`
```
This is not true any more?

Reply:
I only removed the `with HashPartitioning` since we are going to check all partitionings which come from repartition.

Reply:
But if `EnsureRequirements` can't remove user-specified repartition that is not `HashPartitioning`, we don't need to change here?

Reply:
That is right, but it can not describe the following code equivalently.. I changed this comments to:
```
// Note, here are two cases of how user-specified repartition can be optimized out:
// 1. `EnsureRequirements` can only optimize out user-specified repartition with
//    `HashPartitioning`.
// 2. `AQEOptimizer` can optimize out user-specified repartition with all `Partitioning`,
//     e.g. convert empty to local relation.
```